### PR TITLE
test: Add e2e tests

### PR DESCRIPTION
**Pull request by geeksambhu**
## Description
Added e2e tests to cover the Pod Security policy scenarios
These are the end-to-end scenarios covered:

 - Accept all capabilities when star(all) in allowed capabilities
 - Reject when star(all) in allowed capabilities and capabilities in required_drop_capabilities
 - Reject capabilities in required_drop_capabilities
 - Accept capabilities in allowed capabilities
 -  Reject capabilities not listed in allowed capabilities
 - Mutate pods

Fix https://github.com/kubewarden/capabilities-psp-policy/issues/8

## Test

To test this pull request, you can run the following commands:
```shell
make e2e-tests
```

### Potential improvement

Add more scenarios

### CODE REVIEW

**Review by flavio — APPROVED** [review]
LGTM, one minor comment: it would be nice to remove some noise from the `json` file added. There's quite some bloat inside of the files that is not needed by the e2e tests and makes reading the file a bit hard.

File: src/validate.rs [hunk 1]
```diff
@@ -36,7 +36,7 @@ pub(crate) fn validate_added_caps(validation_req: &ValidationRequest<Settings>)
 
     if !must_be_dropped.is_empty() {
         return Err(anyhow!(
-            "PSP capabilities policies doesn't allow these capabilities to be added because they are no the `required_drop_capabilities` list: {:?}",
+            "PSP capabilities policies doesn't allow these capabilities to be added because they are on the `required_drop_capabilities` list: {:?}",
```
Comment (jvanz) [inline src/validate.rs:39]: :+1:

**Review by jvanz — APPROVED** [review]

**Comment by viccuad** [comment]
Many thanks folks for the review!

I agree with Flavio. @geeksambhu , can we reduce the content of the test_data jsons?
It should be enough if they are pod requests, like [this one for example](https://github.com/kubewarden/verify-image-signatures/blob/main/test_data/pod_creation_unsigned.json), and also containing the relevant part of the pod spec for this policy. So, the [`spec.containers.securityContext`](https://github.com/kubewarden/capabilities-psp-policy/pull/36/files#diff-ac3f730db4f21a5e424f6fa98bea47dbc8406c2a73ffd35abadc9e3b1c621c3eR80-R86).

**Comment by geeksambhu** [comment]
Hi @viccuad & @jvanz , I updated the content of test data. Please review

**Review by viccuad — APPROVED** [review]
LGTM! thanks!

**Pull request merged**